src/Directory.Build.targets

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project InitialTargets="AddSkipGetTargetFrameworkToProjectReferences">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), dir.targets))\dir.targets" />
```

Comment:
What are these targets for?

Reply:
This is just what imports the parent directory's dir.targets. We used to have this per project, and we now have it only once on the common file.

Reply:
Should we instead of import dir.targets, rename dir.targets to Directory.Build.Targets?

Reply:
I would be fine with that, though it would probably confuse some devs at first

Reply:
Directory.Build.Targets is a new feature of msbuild that some repos under dotnet/aspnet have started to use already:
https://github.com/search?q=org:dotnet+filename:directory.build.targets
https://github.com/search?q=org:aspnet+filename:directory.build.targets

Reply:
I do think we should rename dir.targets and dir.props to Directory.Build.props/targets.

Reply:
Do you think I should rename all dir.props/dir.targets to Directory.Build.props/targets? @joperezr is there documentation on how SDK projects automatically find files named as such? Will there be collisions if multiple files share that name across different parts of the directory tree?

Reply:
> is there documentation on how SDK projects automatically find files named as such?

Check out https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build

Reply:
You can import the SDK in a central props/targets, but then the central props/targets can't be Directory.Build.*. Chicken and egg. Sdk imports common targets and common targets imports Directory.Build.*. 

If you use explicit dir.props/dir.targets, then you can do this: https://github.com/nguerrera/sdkimportrepro/commit/6d34578b41c4f4e68ec1aa3dace648923be75287

Reply:
dir.props and dir.targets are maltilayered in many repos (they are present at multiple directory levels and recursively each one includes higher level projects). Directory.Build.props/.targets are included once - after first include higher level ones will be skipped. This may cause some problems unless recursive include logic from dirs.props is applied to Directory.Build.* files as well.

Reply:
Yes, this is expected with Directory.Build.*.  It gets imported once, and then if there are parent props/targets files, they need to be included up the chain.  This allows props/target authors to get the ordering right, which is important especially in props files.

We do this in the dotnet/machinelearning repo: https://github.com/dotnet/machinelearning/blob/master/src/Directory.Build.props#L3